tests/ConnectionTest.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Mysql\Tests;
+
+use Yiisoft\Db\Drivers\Connection;
+use Yiisoft\Db\Helper\Dsn;
+use Yiisoft\Db\Tests\ConnectionTest as AbstractConnectionTest;
+
+class ConnectionTest extends AbstractConnectionTest
+{
+    protected ?string $driverName = 'mysql';
+
+    public function testDsnHelper(): void
+    {
+        $dsn = new Dsn('mysql', '127.0.0.1', 'yiitest', '3306');
+
+        $connection = new Connection($this->cache, $this->logger, $this->profiler, $dsn->getDsn());
```

Comment:
It there is need to check the result of `getDsn()` of `Connection`? I think we should check only `Dsn` class.

Reply:
The test checks both.

Reply:
Why? `Connection` just pass any `string|null` to `$dsn` property. We no need to check this.

Reply:
In the constructor accepts only string dsn, and for the purposes of initializing the variable I prefer null which means that it has no status, to assign a blank string.

```php
public function __construct(?CacheInterface $cache, LoggerInterface $logger, Profiler $profiler, string $dsn)
```

Reply:
I am about that Connection is just unnecessary to test `Dsn::getDsn`